test/etc/tasks/ext/phpunit/build.xml

```diff
@@ -1,69 +1,117 @@
+            <formatter type="plain" usefile="false"/>
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="HelloWorldTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+    
+    <target name="testCloverFormatter">
+        <phpunit codecoverage="true" haltonerror="true" haltonfailure="true" printsummary="false">
+            <formatter type="clover" todir="${output.dir}"/>
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="HelloWorldTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+    
+    <target name="testCloverHtmlFormatter">
+        <phpunit codecoverage="true" haltonerror="false" haltonfailure="true" printsummary="true">
+            <formatter type="clover-html" todir="${output.dir}"/>
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="HelloWorldTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+    
+    <target name="testDoubleAutoloader" depends="testAutoloader1,testAutoloader2"/>
+    
+    <target name="testAutoloader1">
+        <phpunit bootstrap="${project.basedir}/bootstrap.php">
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="**/*Test.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+
+    <target name="testAutoloader2">
+        <phpunit bootstrap="${project.basedir}/bootstrap.php">
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="**/*Test.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+    
+    <target name="testShouldNotIgnoreFailuresAfterError">
+        <phpunit haltonerror="false" haltonfailure="true">
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="ErroringTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+    
+    <target name="testExcludeGroups">
+        <phpunit haltonfailure="true" excludeGroups="shouldBeExcluded">
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="ExcludedTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+
+    <target name="testMissingPhpunitConfig">
+        <phpunit configuration="phpunit-missing.xml" codecoverage="false">
+            <formatter type="summary" usefile="false"/>
+            <batchtest>
+                <fileset dir="tests">
+                    <include name="HelloWorldTest.php"/>
+                </fileset>
+            </batchtest>
+        </phpunit>
+    </target>
+
+    <target name="testPhpunitConfig">
+        <phpunit configuration="phpunit-test.xml" codecoverage="false" >
```

Comment:
This will increase test execution time in build from 35s to 61s.

Reply:
> but also adds to the code coverage and specifically tests new code that was added.

To be honest - it just executes the whole test suite again and ignores code-coverage entirely! (`codecoverage="false"`)